### ISSUE TITLE
Can not recover from disconnection

### ISSUE THREAD

**Issue by windmemory** [botorange/wechaty-puppet-padpro#11]
```shell
15:49:53 SILL WechatGateway sendLong() receive back package size: 20
15:49:59 SILL WechatGateway sendLong() reqSeq: 1817798585
15:49:59 SILL WechatGateway sendLong() receive back package size: 47
15:49:59 INFO PadproManager Logged out on phone or log into another device.
15:49:59 WARN PuppetPadpro startManager() manager.on(reset) for undefined. Restarting PuppetPadpro ...
15:49:59 VERB PuppetPadpro stop()
15:49:59 VERB PuppetPadpro logout()
15:49:59 VERB PadproManager logout()
15:49:59 VERB PadproManager releaseCache()
15:49:59 SILL PadproManager releaseCache() closing caches ...
Logout: Contact<我爱抓娃娃-抓抓>
15:49:59 SILL PadproManager releaseCache() cache closed.
15:49:59 VERB PadproManager startCheckScan()
15:49:59 SILL PadproManager startCheckScan() checkScanInternalLoop()
15:49:59 VERB PadproManager emitLoginQrCode()
15:49:59 SILL PadproManager startCheckScan() checkScanInternalLoop() set
15:49:59 VERB PadproManager stop()
15:49:59 WARN PadproManager releaseQueue() subscriptions have been released.
15:49:59 WARN PadproManager releaseQueue() queues have been released.
15:49:59 VERB PadproManager stopCheckScan()
15:49:59 VERB PadproGrpc stop()
15:49:59 VERB PadproManager releaseCache()
15:49:59 VERB PadproManager releaseCache() cache not exist.
15:49:59 VERB PuppetPadpro start() with test
15:49:59 SILL PadproGrpc constructor(puppet_padpro_f100002734f8679a)
15:49:59 VERB PadproManager constructor()
15:49:59 VERB WechatGateway on(newMessage, function) registered
15:49:59 VERB WechatGateway on(rawMessage, function) registered
15:49:59 VERB WechatGateway on(reset, function) registered
15:49:59 VERB PuppetPadpro startManager()
15:49:59 VERB PadproGrpc start()
15:49:59 VERB PadproManager start()
15:49:59 VERB PadproManager tryAutoLogin()
15:49:59 INFO WechatGateway initLongSocket() connected to wechat long host server: long.weixin.qq.com.
15:49:59 SILL WechatGateway sendLong() reqSeq: 1082769055
15:49:59 SILL WechatGateway sendLong() reqSeq: 989765937
15:49:59 SILL PadproManager throttleQueue emit heartbeat.
15:49:59 SILL WechatGateway sendLong() receive back package size: 669
15:50:00 SILL WechatGateway sendLong() receive back package size: 16142
15:50:19 SILL PadproManager debounceQueue heartbeat result: {}
(node:47229) UnhandledPromiseRejectionWarning: Error: 1 CANCELLED: Received http2 header with status: 502
    at Object.exports.createStatusError (/Users/yuangao/git/wechaty/wechaty-puppet-padpro/node_modules/grpc/src/common.js:87:15)
    at Object.onReceiveStatus (/Users/yuangao/git/wechaty/wechaty-puppet-padpro/node_modules/grpc/src/client_interceptors.js:1188:28)
    at InterceptingListener._callNext (/Users/yuangao/git/wechaty/wechaty-puppet-padpro/node_modules/grpc/src/client_interceptors.js:564:42)
    at InterceptingListener.onReceiveStatus (/Users/yuangao/git/wechaty/wechaty-puppet-padpro/node_modules/grpc/src/client_interceptors.js:614:8)
    at callback (/Users/yuangao/git/wechaty/wechaty-puppet-padpro/node_modules/grpc/src/client_interceptors.js:841:24)
(node:47229) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:47229) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
15:50:24 SILL WechatGateway sendLong() reqSeq: 1000418636
15:50:24 SILL PadproManager throttleQueue emit heartbeat.
15:50:24 SILL WechatGateway sendLong() receive back package size: 20
15:50:44 SILL PadproManager debounceQueue heartbeat result: {}
15:50:45 SILL WechatGateway sendLong() reqSeq: 2654373919
15:50:45 SILL PadproManager throttleQueue emit heartbeat.
15:50:45 SILL WechatGateway sendLong() receive back package size: 20
15:50:59 VERB PadproManager tryAutoLogin() failed: Error: 1 CANCELLED: Received http2 header with status: 502
15:50:59 VERB PadproManager startCheckScan()
15:50:59 SILL PadproManager startCheckScan() checkScanInternalLoop()
15:50:59 VERB PadproManager emitLoginQrCode()
15:50:59 SILL PadproManager startCheckScan() checkScanInternalLoop() set
15:50:59 VERB PuppetPadpro startWatchdog()
{ Error: 1 CANCELLED: Received http2 header with status: 502
    at Object.exports.createStatusError (/Users/yuangao/git/wechaty/wechaty-puppet-padpro/node_modules/grpc/src/common.js:87:15)
    at Object.onReceiveStatus (/Users/yuangao/git/wechaty/wechaty-puppet-padpro/node_modules/grpc/src/client_interceptors.js:1188:28)
    at InterceptingListener._callNext (/Users/yuangao/git/wechaty/wechaty-puppet-padpro/node_modules/grpc/src/client_interceptors.js:564:42)
    at InterceptingListener.onReceiveStatus (/Users/yuangao/git/wechaty/wechaty-puppet-padpro/node_modules/grpc/src/client_interceptors.js:614:8)
    at callback (/Users/yuangao/git/wechaty/wechaty-puppet-padpro/node_modules/grpc/src/client_interceptors.js:841:24)
  code: 1,
  metadata: Metadata { _internal_repr: {} },
  details: 'Received http2 header with status: 502' }
```

**Comment by windmemory** [comment]
This issue is about `wechaty-puppet-padpro` can not recover from disconnection.

**Comment by windmemory** [comment]
Should be fixed with 78850ead0b86998142a78653e13b591733020d55
